File: src/Medium/22. Generate Parentheses/ai_solution_3.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        if n == 0:
            return ['']

        dp = [[] for _ in range(n + 1)]
        dp[0] = ['']

        for i in range(1, n + 1):
            for j in range(i):
                for left in dp[j]:
                    for right in dp[i - j - 1]:
                        dp[i].append(f'({left}){right}')

        return dp[n]

File: src/Easy/1. Two Sum/rust/src/ai_solution_3.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut nums_with_indices: Vec<(i32, usize)> = nums.iter().cloned().zip(0..).collect();
        nums_with_indices.sort_unstable();

        let (mut left, mut right) = (0, nums_with_indices.len() - 1);
        while left < right {
            let sum = nums_with_indices[left].0 + nums_with_indices[right].0;
            if sum == target {
                return vec![nums_with_indices[left].1 as i32, nums_with_indices[right].1 as i32];
            } else if sum < target {
                left += 1;
            } else {
                right -= 1;
            }
        }
        vec![]
    }
}